src/ui/common/src/components/resources/dialogs/gcpDialog.tsx

```diff
@@ -1,31 +1,260 @@
-import React from 'react';
+import Box from '@mui/material/Box';
+import Link from '@mui/material/Link';
+import Typography from '@mui/material/Typography';
+import React, { useState } from 'react';
+import { useFormContext } from 'react-hook-form';
 import * as Yup from 'yup';
 
-import { ResourceDialogProps } from '../../../utils/resources';
+import {
+  DynamicEngineType,
+  FileData,
+  GCPConfig,
+  OndemandGKEConfig,
+  ResourceDialogProps,
+} from '../../../utils/resources';
+import { Tab, Tabs } from '../../primitives/Tabs.styles';
+import { ResourceFileUploadField } from './ResourceFileUploadField';
 import { ResourceTextInputField } from './ResourceTextInputField';
 
-// Placeholder component for the GCP dialog.
-export const GCPDialog: React.FC<ResourceDialogProps> = ({
-  editMode = false,
-}) => {
+const Placeholders: OndemandGKEConfig = {
+  gcp_config_serialized: '',
+  keepalive: '1200',
+  cpu_node_type: 'n1-standard-4',
+  gpu_node_type: 'g2-standard-4',
+  min_cpu_node: '1',
+  max_cpu_node: '1',
+  min_gpu_node: '0',
+  max_gpu_node: '1',
+};
+
+const GCPPlaceholders: GCPConfig = {
+  region: 'us-central1',
+  zone: 'us-central1-a',
+  service_account_key: '',
+};
+
+export const GCPDialog: React.FC<ResourceDialogProps> = () => {
```

Comment:
similarly, update with changes in #1380 and follow a few pattern updates. The main necessary thing is to add a type to `ResourceDialogProps` here